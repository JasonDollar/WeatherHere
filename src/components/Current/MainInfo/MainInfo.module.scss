.MainInfo {
  font-size: var(--head-font-size);
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: stretch;
  flex-direction: column;
  margin-top: 1rem;

  @media only screen and (min-width: 576px) {
    flex-direction: row;
    align-items: center;
    flex-wrap: wrap;
  }

  & svg {
    height: var(--head-font-size);
    width: var(--head-font-size);
    transform: scale(2) translateY(-5%);
    vertical-align: middle;
    // overflow: hidden;
  }  
}



.weather {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  // margin: 0 auto;
  flex: 1;
  
  @media only screen and (min-width: 768px) {
    flex: 1 1 48%;
    // flex-wrap: wrap;
  }
  & .city {
    font-size: var(--head-font-size);
    letter-spacing: 1px;
    flex: 1 1 40%;
    margin: 0 1rem 0 2rem;
    text-align: center;
    @media only screen and (min-width: 420px) {
      font-size: var(--big-head-font-size);
    }
  }
  & .temperature {
    flex: 0 1 25%;
    margin: 0 1rem 0 2rem;
    & span {
      font-weight: 300;
    }
  }

  & .icon {
    flex: 0 1 20%;
    text-align: center;
    transform: translateY(10%);
  }
}


.timeInfo {
  display: flex;
  align-items: center;
  justify-content: space-around;
  font-size: var(--big-text-font-size);
  font-weight: 300;
  margin: .5rem 2rem;
  @media only screen and (min-width: 768px) {
    flex: 1 0 45%;
    // flex-wrap: wrap;
  }
  & .date {
    display: flex;
    margin: 0 1rem;
    @media only screen and (min-width: 420px) {
      font-size: var(--small-head-font-size);
    }
    @media only screen and (min-width: 870px) {
      font-size: var(--head-font-size);
    }
  }
  
  & .sunContainer {
    & .sun {
      & span {
        display: inline-block;
        margin-left: 1rem;
        margin-top: 2px;
      }

      & svg {
        height: var(--big-text-font-size);
        width: var(--big-text-font-size);
        transform: scale(2.5) translateY(-5%);
      }

    }
  }
}

.summary {
  @media only screen and (min-width: 420px) {
    // font-size: var(--small-head-font-size);
    // flex-wrap: wrap;
  }
  @media only screen and (min-width: 768px) {
    flex: 0 0 100%;
    font-size: var(--small-head-font-size);
    // flex-wrap: wrap;
  }
  font-size: var(--big-text-font-size);
  margin: .5rem 2rem;
  text-align: center;
}