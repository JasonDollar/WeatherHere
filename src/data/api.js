const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

