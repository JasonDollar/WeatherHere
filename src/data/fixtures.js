import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}