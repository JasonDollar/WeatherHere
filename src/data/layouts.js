const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



