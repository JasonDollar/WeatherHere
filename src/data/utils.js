import moment from 'moment-timezone'

const getTimeFromSeconds = (timestamp, timezone) => {
  const timestampInMiliSec = timestamp * 1000
  if (timezone === null) {
    return {
      month: moment(timestampInMiliSec).format('D.M'),
      // day === week day
      day: moment(timestampInMiliSec).format('e'),
      
      hour: moment.tz(timestampInMiliSec, timezone).format('HH:mm'),
      
    } 
  }
  return {
    month: moment.tz(timestampInMiliSec, timezone).format('D.M'),
    // day === week day
    day: moment.tz(timestampInMiliSec, timezone).format('d'),
    hour: moment.tz(timestampInMiliSec, timezone).format('HH:mm'),
  } 
}

const formatNumber = number => Math.floor(number * 10) / 10


export { getTimeFromSeconds, formatNumber }