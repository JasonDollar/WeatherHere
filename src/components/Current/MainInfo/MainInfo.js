import React from 'react'

import Icon from '../../Icon/Icon'

import { formatNumber, getTimeFromSeconds } from '../../../data/utils'
import classes from './MainInfo.module.scss'


const MainInfo = ({
  locationShortName, currentText, currently, dateText, today, timezone, sunPositionTime,
}) => (
  <div className={classes.MainInfo}>
    <div className={classes.weather}>
      <div className={classes.city}>{locationShortName || currentText.location}</div>
      <div className={classes.icon}>
        <Icon icon={currently.icon} viewBox="0 0 100 100" />  
      </div>   
      <div className={classes.temperature}>
        {/*<Icon icon="temp" viewBox="0 0 100 100" />*/}
        {formatNumber(currently.temperature)}
        <span>Â°C</span>
      </div>
      
    </div>
    <div className={classes.timeInfo}>
      <div className={classes.date}>
        {dateText.weekDay[today.day]} 
        {' '}
        {today.month}
      </div>
      <div className={classes.sunContainer}>
        <div className={classes.sun}>
          <Icon icon="sunrise" viewBox="0 0 100 100" />
          <span>{getTimeFromSeconds(sunPositionTime.sunrise, timezone).hour}</span>
        </div>
        <div className={classes.sun}>
          <Icon icon="sunset" viewBox="0 0 100 100" />
          <span>{getTimeFromSeconds(sunPositionTime.sunset, timezone).hour}</span>
        </div>
      </div>
    </div>
    <p className={classes.summary}>{currently.summary}</p>
  </div>

      

)

export default MainInfo
