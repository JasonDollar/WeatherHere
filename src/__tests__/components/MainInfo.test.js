import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { appTestData as data } from '../../data/fixtures'
import MainInfo from '../../components/Current/MainInfo/MainInfo'
import { formatNumber, getTimeFromSeconds } from '../../data/utils'


let locationShortName; let currentText; let currently; let dateText; let today; let timezone; let sunPositionTime; let units
let wrapper

beforeAll(() => {
  locationShortName = data.locationShortName
  currentText = data.text.pl.current
  currently = data.weather.currently
  dateText = data.text.pl.date
  today = data.today
  timezone = data.timezone
  sunPositionTime = {
    sunrise: 1000,
    sunset: 1001,
  }
  units = data.units.si.id

  wrapper = shallow(
    <MainInfo 
      locationShortName={locationShortName}
      currentText={currentText}
      currently={currently}
      dateText={dateText}
      today={today}
      timezone={timezone}
      sunPositionTime={sunPositionTime}
      units={units}
    />,
  )
})

describe('Main Info component tests', () => {


  test('Matches snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  test('Proper city name rendered', () => {
    const element = wrapper.find('.city')
    expect(element.render().text()).toBe(locationShortName)
  })

  test('Proper temperature rendered', () => {
    const temperature = formatNumber(currently.temperature)
    const temperatureUnits = data.units.si.temperature
    const element = wrapper.find('.temperature')
    expect(element.render().text()).toBe(`${temperature}${temperatureUnits}`)
  })
  test('Proper today date rendered', () => {
    const screenReaderText = currentText.date
    const dayName = dateText.weekDay[today.day]
    const todayDate = today.month
    const element = wrapper.find('.date')
    expect(element.render().text()).toBe(`${screenReaderText} ${dayName} ${todayDate}`)
  })
  test('rendered sunrise section with proper information', () => {
    const sunrise = getTimeFromSeconds(sunPositionTime.sunrise, timezone).hour
    const sunriseText = currentText.sunrise

    const screenReaderElement = wrapper.find('.sun span').at(0)
    const sunriseElement = wrapper.find('.sun span').at(1)
    const sunriseIcon = wrapper.find('.sun Icon').at(0)

    expect(screenReaderElement.text()).toBe(sunriseText)
    expect(sunriseElement.render().text()).toBe(sunrise)
    expect(sunriseIcon.props().icon).toBe('sunrise')
  })
  test('proper sunset time rendered', () => {
    const sunset = getTimeFromSeconds(sunPositionTime.sunset, timezone).hour
    const sunsetText = currentText.sunset

    const screenReaderElement = wrapper.find('.sun span').at(2)
    const sunsetElement = wrapper.find('.sun span').at(3)
    const sunsetIcon = wrapper.find('.sun Icon').at(1)

    expect(screenReaderElement.text()).toBe(sunsetText)
    expect(sunsetElement.render().text()).toBe(sunset)
    expect(sunsetIcon.props().icon).toBe('sunset')
  })

  test('proper props given to Summary component from MainInfo comp.', () => {
    const element = wrapper.find('Summary')
    expect(element.props().children).toBe(currently.summary)
  }) 
  test('proper props icon name given to Icon component from MainInfo comp.', () => {
    const element = wrapper.find('.icon Icon')
    expect(element.props().icon).toBe(currently.icon)
  }) 
})