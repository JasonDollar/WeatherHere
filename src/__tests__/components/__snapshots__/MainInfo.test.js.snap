// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Info component tests MainInfo to match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainInfo
    currentText={
      Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      }
    }
    currently={
      Object {
        "apparentTemperature": 6.53,
        "cloudCover": 0.92,
        "dewPoint": 8.45,
        "humidity": 0.94,
        "icon": "partly-cloudy-night",
        "ozone": 290.56,
        "precipIntensity": 0.0076,
        "precipProbability": 0.01,
        "precipType": "rain",
        "pressure": 1009.97,
        "summary": "Średnie zachmurzenie",
        "temperature": 9.38,
        "time": 1540834485,
        "uvIndex": 0,
        "visibility": 16.09,
        "windBearing": 99,
        "windGust": 10.76,
        "windSpeed": 5.76,
      }
    }
    dateText={
      Object {
        "temperature": "Temperatura",
        "temperatureMax": "Temperatura Max.",
        "temperatureMin": "Temperatura Min.",
        "today": "Dziś",
        "weekDay": Array [
          "Niedziela",
          "Poniedziałek",
          "Wtorek",
          "Środa",
          "Czwartek",
          "Piątek",
          "Sobota",
          "Dziś",
        ],
        "weekDayShort": Array [
          "Nie",
          "Pon",
          "Wt",
          "Śr",
          "Czw",
          "Pt",
          "Sob",
        ],
      }
    }
    locationShortName="Plock"
    sunPositionTime={
      Object {
        "sunrise": 1000,
        "sunset": 1001,
      }
    }
    timezone="Europe/Warsaw"
    today={
      Object {
        "day": "1",
        "hour": "11:11",
        "month": "1.1",
      }
    }
    units="si"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="weather"
        >
          <div
            className="city"
          >
            Plock
          </div>
          <div
            className="icon"
          >
            <Icon
              icon="partly-cloudy-night"
              title="Ikona pogody"
              viewBox="0 0 100 100"
            />
          </div>
          <div
            className="temperature"
          >
            9.3
            <span>
              °C
            </span>
          </div>
        </div>,
        <div
          className="timeInfo"
        >
          <div
            className="date"
          >
            <span
              className="hidden"
            >
              Data
               
            </span>
            Poniedziałek
             
            1.1
          </div>
          <div
            className="sunContainer"
          >
            <div
              className="sun"
            >
              <span
                className="hidden"
              >
                Wschód słońca
              </span>
              <Icon
                icon="sunrise"
                title="Słońce"
                viewBox="0 0 100 100"
              />
              <span>
                01:16
              </span>
            </div>
            <div
              className="sun"
            >
              <span
                className="hidden"
              >
                Zachód słońca
              </span>
              <Icon
                icon="sunset"
                title="Słońce"
                viewBox="0 0 100 100"
              />
              <span>
                01:16
              </span>
            </div>
          </div>
        </div>,
        <Summary>
          Średnie zachmurzenie
        </Summary>,
      ],
      "className": "MainInfo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="city"
            >
              Plock
            </div>,
            <div
              className="icon"
            >
              <Icon
                icon="partly-cloudy-night"
                title="Ikona pogody"
                viewBox="0 0 100 100"
              />
            </div>,
            <div
              className="temperature"
            >
              9.3
              <span>
                °C
              </span>
            </div>,
          ],
          "className": "weather",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Plock",
              "className": "city",
            },
            "ref": null,
            "rendered": "Plock",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                icon="partly-cloudy-night"
                title="Ikona pogody"
                viewBox="0 0 100 100"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "icon": "partly-cloudy-night",
                "title": "Ikona pogody",
                "viewBox": "0 0 100 100",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                9.3,
                <span>
                  °C
                </span>,
              ],
              "className": "temperature",
            },
            "ref": null,
            "rendered": Array [
              9.3,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "°C",
                },
                "ref": null,
                "rendered": "°C",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="date"
            >
              <span
                className="hidden"
              >
                Data
                 
              </span>
              Poniedziałek
               
              1.1
            </div>,
            <div
              className="sunContainer"
            >
              <div
                className="sun"
              >
                <span
                  className="hidden"
                >
                  Wschód słońca
                </span>
                <Icon
                  icon="sunrise"
                  title="Słońce"
                  viewBox="0 0 100 100"
                />
                <span>
                  01:16
                </span>
              </div>
              <div
                className="sun"
              >
                <span
                  className="hidden"
                >
                  Zachód słońca
                </span>
                <Icon
                  icon="sunset"
                  title="Słońce"
                  viewBox="0 0 100 100"
                />
                <span>
                  01:16
                </span>
              </div>
            </div>,
          ],
          "className": "timeInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="hidden"
                >
                  Data
                   
                </span>,
                "Poniedziałek",
                " ",
                "1.1",
              ],
              "className": "date",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Data",
                    " ",
                  ],
                  "className": "hidden",
                },
                "ref": null,
                "rendered": Array [
                  "Data",
                  " ",
                ],
                "type": "span",
              },
              "Poniedziałek",
              " ",
              "1.1",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="sun"
                >
                  <span
                    className="hidden"
                  >
                    Wschód słońca
                  </span>
                  <Icon
                    icon="sunrise"
                    title="Słońce"
                    viewBox="0 0 100 100"
                  />
                  <span>
                    01:16
                  </span>
                </div>,
                <div
                  className="sun"
                >
                  <span
                    className="hidden"
                  >
                    Zachód słońca
                  </span>
                  <Icon
                    icon="sunset"
                    title="Słońce"
                    viewBox="0 0 100 100"
                  />
                  <span>
                    01:16
                  </span>
                </div>,
              ],
              "className": "sunContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="hidden"
                    >
                      Wschód słońca
                    </span>,
                    <Icon
                      icon="sunrise"
                      title="Słońce"
                      viewBox="0 0 100 100"
                    />,
                    <span>
                      01:16
                    </span>,
                  ],
                  "className": "sun",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Wschód słońca",
                      "className": "hidden",
                    },
                    "ref": null,
                    "rendered": "Wschód słońca",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "icon": "sunrise",
                      "title": "Słońce",
                      "viewBox": "0 0 100 100",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "01:16",
                    },
                    "ref": null,
                    "rendered": "01:16",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="hidden"
                    >
                      Zachód słońca
                    </span>,
                    <Icon
                      icon="sunset"
                      title="Słońce"
                      viewBox="0 0 100 100"
                    />,
                    <span>
                      01:16
                    </span>,
                  ],
                  "className": "sun",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Zachód słońca",
                      "className": "hidden",
                    },
                    "ref": null,
                    "rendered": "Zachód słońca",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "icon": "sunset",
                      "title": "Słońce",
                      "viewBox": "0 0 100 100",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "01:16",
                    },
                    "ref": null,
                    "rendered": "01:16",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Średnie zachmurzenie",
        },
        "ref": null,
        "rendered": "Średnie zachmurzenie",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="weather"
          >
            <div
              className="city"
            >
              Plock
            </div>
            <div
              className="icon"
            >
              <Icon
                icon="partly-cloudy-night"
                title="Ikona pogody"
                viewBox="0 0 100 100"
              />
            </div>
            <div
              className="temperature"
            >
              9.3
              <span>
                °C
              </span>
            </div>
          </div>,
          <div
            className="timeInfo"
          >
            <div
              className="date"
            >
              <span
                className="hidden"
              >
                Data
                 
              </span>
              Poniedziałek
               
              1.1
            </div>
            <div
              className="sunContainer"
            >
              <div
                className="sun"
              >
                <span
                  className="hidden"
                >
                  Wschód słońca
                </span>
                <Icon
                  icon="sunrise"
                  title="Słońce"
                  viewBox="0 0 100 100"
                />
                <span>
                  01:16
                </span>
              </div>
              <div
                className="sun"
              >
                <span
                  className="hidden"
                >
                  Zachód słońca
                </span>
                <Icon
                  icon="sunset"
                  title="Słońce"
                  viewBox="0 0 100 100"
                />
                <span>
                  01:16
                </span>
              </div>
            </div>
          </div>,
          <Summary>
            Średnie zachmurzenie
          </Summary>,
        ],
        "className": "MainInfo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="city"
              >
                Plock
              </div>,
              <div
                className="icon"
              >
                <Icon
                  icon="partly-cloudy-night"
                  title="Ikona pogody"
                  viewBox="0 0 100 100"
                />
              </div>,
              <div
                className="temperature"
              >
                9.3
                <span>
                  °C
                </span>
              </div>,
            ],
            "className": "weather",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Plock",
                "className": "city",
              },
              "ref": null,
              "rendered": "Plock",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  icon="partly-cloudy-night"
                  title="Ikona pogody"
                  viewBox="0 0 100 100"
                />,
                "className": "icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": "partly-cloudy-night",
                  "title": "Ikona pogody",
                  "viewBox": "0 0 100 100",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  9.3,
                  <span>
                    °C
                  </span>,
                ],
                "className": "temperature",
              },
              "ref": null,
              "rendered": Array [
                9.3,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "°C",
                  },
                  "ref": null,
                  "rendered": "°C",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="date"
              >
                <span
                  className="hidden"
                >
                  Data
                   
                </span>
                Poniedziałek
                 
                1.1
              </div>,
              <div
                className="sunContainer"
              >
                <div
                  className="sun"
                >
                  <span
                    className="hidden"
                  >
                    Wschód słońca
                  </span>
                  <Icon
                    icon="sunrise"
                    title="Słońce"
                    viewBox="0 0 100 100"
                  />
                  <span>
                    01:16
                  </span>
                </div>
                <div
                  className="sun"
                >
                  <span
                    className="hidden"
                  >
                    Zachód słońca
                  </span>
                  <Icon
                    icon="sunset"
                    title="Słońce"
                    viewBox="0 0 100 100"
                  />
                  <span>
                    01:16
                  </span>
                </div>
              </div>,
            ],
            "className": "timeInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="hidden"
                  >
                    Data
                     
                  </span>,
                  "Poniedziałek",
                  " ",
                  "1.1",
                ],
                "className": "date",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Data",
                      " ",
                    ],
                    "className": "hidden",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Data",
                    " ",
                  ],
                  "type": "span",
                },
                "Poniedziałek",
                " ",
                "1.1",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="sun"
                  >
                    <span
                      className="hidden"
                    >
                      Wschód słońca
                    </span>
                    <Icon
                      icon="sunrise"
                      title="Słońce"
                      viewBox="0 0 100 100"
                    />
                    <span>
                      01:16
                    </span>
                  </div>,
                  <div
                    className="sun"
                  >
                    <span
                      className="hidden"
                    >
                      Zachód słońca
                    </span>
                    <Icon
                      icon="sunset"
                      title="Słońce"
                      viewBox="0 0 100 100"
                    />
                    <span>
                      01:16
                    </span>
                  </div>,
                ],
                "className": "sunContainer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="hidden"
                      >
                        Wschód słońca
                      </span>,
                      <Icon
                        icon="sunrise"
                        title="Słońce"
                        viewBox="0 0 100 100"
                      />,
                      <span>
                        01:16
                      </span>,
                    ],
                    "className": "sun",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Wschód słońca",
                        "className": "hidden",
                      },
                      "ref": null,
                      "rendered": "Wschód słońca",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "icon": "sunrise",
                        "title": "Słońce",
                        "viewBox": "0 0 100 100",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "01:16",
                      },
                      "ref": null,
                      "rendered": "01:16",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="hidden"
                      >
                        Zachód słońca
                      </span>,
                      <Icon
                        icon="sunset"
                        title="Słońce"
                        viewBox="0 0 100 100"
                      />,
                      <span>
                        01:16
                      </span>,
                    ],
                    "className": "sun",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Zachód słońca",
                        "className": "hidden",
                      },
                      "ref": null,
                      "rendered": "Zachód słońca",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "icon": "sunset",
                        "title": "Słońce",
                        "viewBox": "0 0 100 100",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "01:16",
                      },
                      "ref": null,
                      "rendered": "01:16",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Średnie zachmurzenie",
          },
          "ref": null,
          "rendered": "Średnie zachmurzenie",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
